minio/minio
# Fix replication regression

From PR #11165 due to incorrect
proxying for 2 way replication even
when the object was not yet replicated

Also fixing metadata comparisons when
deciding to do full replciation vs metadata copy.

Also Fixes #11340

## Description


## Motivation and Context


## How to test this PR?
Turn on 2 way replication and do some uploads to source. `mc admin trace` should show a PUT operation on the dest, not a COPY in master due to incorrect proxying.

## Types of changes

## Checklist:
